modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/model/Permission.kt

```diff
@@ -0,0 +1,88 @@
+package net.twisterrob.ghlint.model
+
+import java.util.Locale
+
+// https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+public interface Permissions {
+	public val actions: Access
+	public val attestations: Access
+	public val checks: Access
+	public val contents: Access
+	public val deployments: Access
+	public val idToken: Access
+	public val issues: Access
+	public val metadata: Access
+	public val packages: Access
+	public val pages: Access
+	public val pullRequests: Access
+	public val repositoryProjects: Access
+	public val securityEvents: Access
+	public val statuses: Access
+
+	public fun asMap(): Map<String, String>
+}
+
+public enum class Permission(public val value: String) {
+	ACTIONS("actions"),
+	ATTESTATIONS("attestations"),
+	CHECKS("checks"),
+	CONTENTS("contents"),
+	DEPLOYMENTS("deployments"),
+	ID_TOKEN("id-token"),
+	ISSUES("issues"),
+	METADATA("metadata"),
+	PACKAGES("packages"),
+	PAGES("pages"),
+	PULL_REQUESTS("pull-requests"),
```

Comment:
Is this what you meant?

Reply:
yep, just call it `name`

Reply:
`name` was causing issues - something about enums being final and you can't override name? Is that an implicit variable?

Reply:
ah of course, Enum.name is built-in 🤦🏻‍♂️, that's what you were uppercasing.
Navigate to `kotlin.Enum`. Ok, then, `value` it is.